### ISSUE TITLE
Stencil web API not transpiling correctly in 2.7.0

### ISSUE THREAD

**Issue by theo-staizen** [ionic-team/stencil#3011]
**Stencil version:**

```
 @stencil/core@2.7.0
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**
The `transpile` method of `@stencil/core/compiler/stencil.js` is no longer transpiling the source code given.
It seems to be returning the source code back as is, i.e the jsx/typescript is not compiled.

**Expected behavior:**
The transpile method should work the same as previous versions (2.6.0 or earlier).

**Steps to reproduce:**

See this fiddle: https://jsfiddle.net/dogoku/25Lstbfv/

**Related code:**

```tsx
// assuming "https://cdn.jsdelivr.net/npm/@stencil/core@2.7.0/compiler/stencil.min.js" was loaded

const src = `
  import { h } from "https://cdn.skypack.dev/@stencil/core";
  @Component({ tag: 'cmp-a' })
  export class CmpA {
    @Prop() name
    render() {
      return <h1>Hello {this.name}</h1>
    }
  }
`;
const result = window.stencil.transpileSync(src, { sourceMap: false });
console.log(result.code)
```
Using, 2.7.0, `result.code` is the same as `src`. Using 2.6.0, `result.code` should be: 

```js
import { defineCustomElement as __stencil_defineCustomElement } from "@stencil/core/internal/client";
import { h } from "https://cdn.skypack.dev/@stencil/core";
const CmpA = class extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
  }
  render() {
    return h("h1", null,
      "Hello ",
      this.name);
  }
};
__stencil_defineCustomElement(CmpA, [0, "cmp-a", {
    "name": [8]
  }]);
```

**Comment by rwaskiewicz** [comment]
Thanks for the detailed error report and reproduction! We're going to take a look into this during this sprint

**Comment by splitinfinities** [comment]
1381cc7e920d7d9880d046693762b0f2348c8b5d introduces this issue.

**Comment by rwaskiewicz** [comment]
Since I merged the PR that referenced this, GH says I 'closed' the issue (even though it's still valid on the latest release of Stencil, v2.7.0).  I merged @splitinfinities #3012 which fixes this issue. The current plan of action is to release a dev build with the fix to do a little more "real world" testing on Monday. If all goes well, we will cut a proper release later that day.

**Comment by rwaskiewicz** [comment]
We believe this issue has been resolved as of v2.7.1. If it continues to persist, please let us know in a new issue. Thanks again!
